File: w09/d01/matt_lucas/mailer/spec/package_spec.rb
require_relative('../lib/package')

describe Package do
  it "should have a ship_to_address" do
    package = Package.new(ship_to_address: "920 Broadway")
    expect(package.ship_to_address).to eq("920 Broadway")
  end

  it "should have a content array" do
    package = Package.new
    expect(package.contents.class).to eq(Array)
  end

  it "should have an is_fragile boolean" do
    package = Package.new(is_fragile: true)
    expect(package.is_fragile?).to eq(true)
  end

  it "should have an is_broken boolean" do
    package = Package.new(is_broken: false)
    expect(package.is_broken?).to eq(false)
  end

  it "should have a method called shake" do
    package = Package.new
    package.shake
    expect(package.is_broken?).to eq(false)
  end

  it "should break if the shake method is called" do
    package = Package.new(is_fragile: true)
    package.shake
    expect(package.is_broken?).to eq(true)
  end

end

File: w09/d03/jorge_colindres/mail_objects_app/spec/javascripts/package_spec.js
describe('a package', function(){

  var packageObject;
  beforeEach(function(){
    packageObject = new Package('a book');
  });

  it('should have content', function(){
    expect(packageObject.contents).toBeDefined();
    expect(packageObject.contents).toBe('a book');
  });

  it('should have a fragile boolean', function(){
    packageObject.fragile = true;
    expect(packageObject.fragile).toBeTruthy();
    packageObject.fragile = false;
    expect(packageObject.fragile).toBeFalsy();
  });

  it('should have a broken boolean', function(){
    packageObject.broken = true;
    expect(packageObject.broken).toBeTruthy();
    packageObject.broken = false;
    expect(packageObject.broken).toBeFalsy();
  });

  it('should have a method shake', function(){
    expect(packageObject.shake).toBeDefined();
  });

  it('should break if it is fragile and is shaken', function(){
    packageObject.fragile = true;
    packageObject.shake();
    expect(packageObject.broken).toBeTruthy();
  });

});
